src/Select.js

```diff
@@ -1066,6 +1072,51 @@ class Select extends React.Component {
 		);
 	}
 
+	renderUniqueSelect(className, valueArray) {
+		const focusedOptionIndex = this.getFocusableOptionIndex(valueArray[0]);
+		let options = this._visibleOptions = this.filterOptions(this.props.multi ? this.getValueArray(this.props.value) : null);
+		let focusedOption = null;
+		if (focusedOptionIndex !== null) {
+			focusedOption = this._focusedOption = options[focusedOptionIndex];
+		} else {
+			focusedOption = this._focusedOption = null;
+		}
+
+		return (
+			<div ref={ref => this.wrapper = ref}
+					className={className}
+					style={this.props.wrapperStyle}>
+				{this.renderHiddenField(valueArray)}
+				<div className="Select-unique-input-value-wrapper">
+					{this.renderValue(valueArray, true)}
+					{this.renderClear()}
+				</div>
+				<div className="Select-unique-input-list-wrapper">
+					<div ref={ref => this.control = ref}
+							className="Select-control is-unique"
```

Comment:
I think you don't need to add the `is-unique` class here. You can change your SCSS rule to be `.is-unique .Select-control` instead of `.Select-control.is-unique`.